js.go

```diff
@@ -827,11 +828,16 @@ func (m *Msg) checkReply() (*js, bool, error) {
 }
 
 // ackReply handles all acks. Will do the right thing for pull and sync mode.
+// It ensures that an ack is only sent a single time, regardless of
+// how many times it is being called to avoid duplicated acks.
 func (m *Msg) ackReply(ackType []byte, sync bool) error {
 	js, isPullMode, err := m.checkReply()
 	if err != nil {
 		return err
 	}
+	if atomic.LoadUint32(&m.ackd) == 1 {
```

Comment:
If the first ack I do is a progress type I think this will fail if I send anything after that..

Reply:
The test `TestJetStreamSubscribe_AckDupInProgress` think should cover this, but this is what would happen right now:

```
m.InProgress() // Sends Ack to update progress
m.InProgress() // Sends Ack as well
m.Ack()             // Also sends Ack, after this no more acks are sent
m.Ack()             // Does not send Ack
m.InProgress() // Would not send Ack either
```

https://github.com/nats-io/nats.go/pull/637/files#diff-9e6bdee4051a6e36d6d507dcef63b5f9a39b281e421c08f02ec74a0fb6639e3aR1465-R1470

Reply:
Apologies you are right. Was reading that wrong..